### ISSUE TITLE
Manual transformation of the 'data' slot to 'scale.data'

### ISSUE THREAD

**Issue by Roooosh** [ChristophH/sctransform#17]
Hi Christoph,
From the documentation, SCTransform replaces 3 regular steps in the Seurat workflow: NormalizeData, ScaleData and FindVariableFeatures.

I was wondering if it was possible to do these in a step wise fashion. More so, going from the log1p(counts)(found in the data slot) to the pearson residuals(found in the 'scale.data' slot)

I do not think the ScaleData function will output what I need, is there a work around to do this? If so, what do you suggest? Thanks!

**Comment by ChristophH** [comment]
Hi,
I don't understand what it is that you would like to do. Internally `SCTransform()` calls `sctransform::vst()` to transform UMI counts directly to Pearson residuals (`scale.data`). These residuals determine which genes are considered highly variable. Also, if `do.correct.umi = TRUE` the Pearson residuals are turned into `counts` (a if all cells had the same number of UMI). These `counts` are further transformed by `log1p()` to fill the `data` slot.
I hope this explanation helps.

**Comment by Roooosh** [comment]
Ah, I see. I am currently using the SCTransform in the context of the integration workflow.
I noticed that when I performed the integration on data that was SCTransform'd, that within resulting 'integrated' object, the 'scale data' slot in the 'SCT' assay was empty. I wanted to visualize gene expression via a heatmap in this assay and compare it to the 'integrated' object, but was thrown an error that:

'No requested features found in the scale.data slot for the SCT assay.'

From your response, should I just be comparing the 'data' slots of both objects?
